readthedocs/oauth/services/gitlab.py

```diff
@@ -0,0 +1,330 @@
+# -*- coding: utf-8 -*-
+"""OAuth utility functions."""
+
+from __future__ import division, print_function, unicode_literals
+
+import json
+import logging
+import re
+
+from allauth.socialaccount.providers.gitlab.views import GitLabOAuth2Adapter
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from requests.exceptions import RequestException
+
+from readthedocs.builds import utils as build_utils
+from readthedocs.integrations.models import Integration
+
+from ..models import RemoteOrganization, RemoteRepository
+from .base import DEFAULT_PRIVACY_LEVEL, Service
+
+try:
+    from urlparse import urljoin, urlparse
+except ImportError:
+    from urllib.parse import urljoin, urlparse  # noqa
+
+log = logging.getLogger(__name__)
+
+
+class GitLabService(Service):
+
+    """
+    Provider service for GitLab.
+
+    See:
+     - https://docs.gitlab.com/ce/integration/oauth_provider.html
+     - https://docs.gitlab.com/ce/api/oauth2.html
+    """
+
+    adapter = GitLabOAuth2Adapter
+    # Just use the network location to determine if it's a GitLab project
+    # because private repos have another base url, eg. git@gitlab.example.com
+    url_pattern = re.compile(
+        re.escape(urlparse(adapter.provider_base_url).netloc))
+    default_avatar = {
+        'repo': urljoin(settings.MEDIA_URL, 'images/fa-bookmark.svg'),
+        'org': urljoin(settings.MEDIA_URL, 'images/fa-users.svg'),
```

Comment:
Do we want these default avatars? Github and Bitbucket are not using nothing similar to this.

Reply:
I'd say no, we use the github/bitbucket supplied avatar as the only source I believe.

Reply:
I will remove the default avatars and its files.

We can make it in another PR for all the services, but I think it doesn't worth it really.